app/assets/javascripts/mobile.js

```diff
@@ -35,24 +34,68 @@ const tracking = function(params) {
       }
 
       if (event.data.type === 'SYNC_ERROR') {
-        stickyError(I18n.t('errors.sync_failed'));
+        stickyError(I18n.t('errors.mobile.sync_failed'));
+      }
+
+      if (event.data.type === 'STORE_STOPS') {
+        event.data.data.forEach(stop => {
+          localStorage.setItem(`stop_update_${stop.id}`, JSON.stringify(stop));
+        });
+      }
+
+      if (event.data.type === 'STORE_POSITIONS') {
+        event.data.data.forEach(position => {
+          localStorage.setItem(`position_${position.id}`, JSON.stringify(position));
+        });
+      }
+
+      if (event.data.type === 'GET_PENDING_DATA') {
+        const data = {
+          positions: [],
+          stops: []
+        };
+
+        for (let i = 0; i < localStorage.length; i++) {
+          const key = localStorage.key(i);
+          if (key.startsWith('position_')) {
+            const position = JSON.parse(localStorage.getItem(key));
+            data.positions.push(position);
+            localStorage.removeItem(key);
+          } else if (key.startsWith('stop_update_')) {
+            const stop = JSON.parse(localStorage.getItem(key));
+            data.stops.push(stop);
+            localStorage.removeItem(key);
+          }
+        }
+
+        event.source.postMessage({
+          type: 'PENDING_DATA',
+          data: data
+        });
       }
     };
 
     navigator.serviceWorker.addEventListener('message', messageListener);
 
     const token = document.querySelector('meta[name="csrf-token"]').getAttribute('content');
-    navigator.serviceWorker.register('/service-worker.js', {
+    navigator.serviceWorker.register('/service-worker.js?' + Date.now(), {
```

Comment:
No better way than adding `Date.now()` ?

Reply:
Nice catch, it is no more needed in this version as we are retrieving the stored data both on loading and once back online. Requiring the service-worker to keep the data was a misunderstanding of its purpose.